samples/azure-quantum/simulators/SparseSimulator.ipynb

```diff
@@ -0,0 +1,2274 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "acquired-exhibit",
+   "metadata": {},
+   "source": [
+    "# Sparse Simulator\n",
+    "\n",
+    "The sparse simulator is a simulator that utilizes a sparse representation of quantum state vectors, as opposed to the full state simulator. This feature allows the sparse simulator to minimize the memory footprint used to represent quantum states that are sparse in the computational basis, thus enabling simulations over a larger number of qubits. The sparse simulator is efficient for a class of quantum algorithms with a smaller number of states in superposition. This simulator is an unlocker for users to explore larger applications than what can be explored using the full state simulator alone. For more information about the sparse simulator see [this paper](https://arxiv.org/abs/2105.01533).\n",
+    "\n",
+    "Let's see the advantages of the sparse simulator over the [full state simulator](https://docs.microsoft.com/en-us/azure/quantum/user-guide/machines/full-state-simulator).\n",
+    "\n",
+    "Let's create a few groups (clusters) of qubits with GHZ state. Please execute the cell below. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "affiliated-value",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[\"PrepareAndMeasureGhzState\"]",
+      "text/html": [
+       "<ul><li>PrepareAndMeasureGhzState</li></ul>"
+      ],
+      "text/plain": [
+       "PrepareAndMeasureGhzState"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Canon;\n",
+    "open Microsoft.Quantum.Intrinsic;\n",
+    "open Microsoft.Quantum.Arrays;\n",
+    "open Microsoft.Quantum.Diagnostics;\n",
+    "open Microsoft.Quantum.Measurement;\n",
```

Comment:
As is, the notebook jumps right in without explaining any of the code or how it's used. Please break these into a separate cell, and explain what these namespaces provide for user code.